gramps/plugins/tool/reorderids.py

```diff
@@ -475,6 +495,16 @@ def _display(self):
                                     self.obj_values[prim_obj].get_keep,
                                     readonly=True)
 
+            if gtkok:
+                # the max_width_chars function was introduced in Gtk 3.12
+                # for earlier versions the dialog might be too large on small
+                # screens since the default Gtk.Entry seems to be about 20
+                # characters or so.
+                obj_actual.set_max_width_chars(10)
```

Comment:
Use `set_width_chars`, which is available in Gtk 3.10, instead.

Reply:
@Nick-Hall Unfortunately, `set_width_chars` doesn't seem to work by itself, at least on my Windows Gtk 3.18.9.  The original Glade file had (and still has) that, and the Gtk.Entry fields were all very wide (20 chars or so despite `set_width_chars` being set to 10, or 6.  Leading to the ~1000px wide dialog.  I tried quite a few combinations before I found the stackexchange reference to doing it this way.

Reply:
@prculley Interesting.  It works in the filter sidebars to keep them small.

Reply:
@Nick-Hall Attached is the original, unmodified ReorderIds screen from gramps50, running on my kubuntu Linux VM.  Note that even with the `set_width_chars` values that you want, the fields are large.  I also note that my sidebar is very wide as well (that's as small as it gets), I never realized that it wasn't supposed to be that way until I compared with an image from our wiki (also attached), but that is another bug.
So I don't think the issue is Windows specific, but probably is Gtk specific.

So I think the code should stay as it is; at least for me it works well on both Linux and Windows.

![redilinux](https://user-images.githubusercontent.com/18122203/30885657-b9c31ec2-a2d9-11e7-9d0a-6688e18d60e0.jpg)
![gramps_sidebaricons_filter](https://user-images.githubusercontent.com/18122203/30885938-d458d2a8-a2da-11e7-86d6-936363616a15.png)

Reply:
What version of Kubuntu are you using?  I'm using Ubuntu 14.10.

When you just want to set the default width and not the maximum width, using `set_max_width_chars` is not a good idea.  It is probably acceptable here, because all the fields are small.  However, it will cause problems if you try to use the same fix the the gramplet sidebar where users will want to increase the size on large screens.

The window for this tool also cannot be resized, which is inconsistent with other tools in Gramps.

Reply:
I definitely understand that setting max_width_chars means that the field (and thus the whole dialog) becomes locked to no larger than that.  With the Gtk (bug/feature) that the Gtk.Entry doesn't go smaller than 20 chars (or so), despite set_width_chars values smaller than that, the dialog size becomes completely locked.  If Gtk ever allows a smaller Gtk.Entry, the dialog will still shrink.
But for this application I cannot see why anyone would want to resize it anyway.  I deliberately made the fields almost twice the size needed so there should not be an issue even for different size default fonts.

I'm open to suggestions on how to shrink these fields without using max_width_chars; I spent 1/2 a day searching the internet and trying various possibilities already, none worked.  And I really think that having a dialog ~1000px wide is not a good idea.

My Linux KUbuntu system is 15.10 with Gtk 3.16.7, my Windows system (where the original dialog looks exactly the same) has Gtk 3.18.9  which is where Josip has frozen the AIO Gtk for now.

Reply:
I don't see this issue in Gtk 3.10 or Gtk 3.22 - so it looks like a bug that occurred and was fixed between these two versions.

Our policy is that no dialog should be larger than 800x600, so you need the fix for broken Gtk versions.  However, I would not apply the fix to versions that don't need it and I would also include a link to the Gtk bug report in the comment.  We tend to remove fixes when the minimum required Gtk version reaches the point that they are no longer required.

Dialog windows should be made resizable for consistency, even if there isn't much point for them to be made bigger than their default size.